sky/packages/sky/lib/src/fn3/framework.dart

```diff
@@ -162,7 +162,37 @@ abstract class ComponentState<T extends StatefulComponent> {
   /// Returns another Widget out of which this StatefulComponent is built.
   /// Typically that Widget will have been configured with further children,
   /// such that really this function returns a tree of configuration.
-  Widget build();
+  Widget build(BuildContext context);
+}
+
+abstract class ProxyWidget extends StatelessComponent {
+  const ProxyWidget({ Key key, Widget this.child }) : super(key: key);
+
+  final Widget child;
+
+  Widget build(BuildContext context) => child;
+}
+
+abstract class ParentDataWidget extends ProxyWidget {
+  ParentDataWidget({ Key key, Widget child })
+    : super(key: key, child: child);
+
+  /// Subclasses should override this function to ensure that they are placed
+  /// inside widgets that expect them.
+  ///
+  /// The given ancestor is the first RenderObjectWidget ancestor of this widget.
+  void debugValidateAncestor(RenderObjectWidget ancestor);
+
+  void updateParentData(RenderObject renderObject);
```

Comment:
call this one updateParentData() so that it's distinct from the other updateParentData().